src/others/components/FilterDropdown/ListWithSearch.tsx

```diff
@@ -0,0 +1,113 @@
+import { FixedSizeList, areEqual } from "react-window";
+import { Button, ListItem, ListItemButton, ListItemText, ClickAwayListener, Divider, useAutocomplete, Typography } from "@mui/material";
+import { FunctionComponent, memo, forwardRef } from "react";
+import SearchIcon from "@mui/icons-material/Search";
+
+import { FilterItem } from "../../contexts/filter";
+
+import styles from "./ListWithSearch.module.css";
+
+type ListWithSearchProps = {
+  selectedFilterItemCount: number;
+  selectedFilterItems: FilterItem[];
+  searchableItems: FilterItem[];
+  onSelectItem: (id: string, selected: boolean) => () => void;
+  checkboxListItemIcon: (value: boolean) => React.ReactNode;
+  toggleFilterList: () => void;
+  clearAllFilters: () => void;
+};
+
+function itemKey(index: number, data: FilterItem[]) {
+  const item = data[index];
+  return `${item.id}-${index}`;
```

Comment:
why simply item.id cannot be used as a key?

Reply:
There are some duplicate city names, somewhat tangential to https://github.com/ukraine-taskforce/requests-dashboard/pull/55#discussion_r831104778

Reply:
sure, that's why i recommended using numeric id which is unique

Reply:
It's unique, but it's unreadable - that's why it needs to be `${item.id}-${index}`

Reply:
why do you need to "read" them? :) you read the name not the id